service/src/main/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperService.java

```diff
@@ -145,23 +151,32 @@ private boolean respondingOnBehalfIsEqual(DirectionResponse response, Element<Di
      * @param directionsMap a map where the DirectionAssignee key corresponds to a list of directions elements.
      */
     public void addDirectionsToCaseDetails(CaseDetails caseDetails,
-                                           Map<DirectionAssignee, List<Element<Direction>>> directionsMap) {
+                                           Map<DirectionAssignee, List<Element<Direction>>> directionsMap,
+                                           ComplyOnBehalfEvent eventId) {
         directionsMap.forEach((assignee, directions) -> {
             final List<Element<Direction>> clone = getClone(directionsMap.get(ALL_PARTIES));
 
             switch (assignee) {
                 case PARENTS_AND_RESPONDENTS:
                     directions.addAll(clone);
 
-                    filterResponsesNotCompliedOnBehalfOfByTheCourt("RESPONDENT", directions);
+                    if (eventId == COMPLY_ON_BEHALF_SDO) {
```

Comment:
If you do agree to go with the above change to `EventID` for the enum then this must either a switch statement or have the `else` become an `else if`.